projects/admin/src/app/record/brief-view/loans-brief-view/loans-brief-view.component.ts

```diff
@@ -0,0 +1,134 @@
+  patronUrl: string = null;
+  /** class to use for the state bullet */
+  stateClass: string = null;
+  /** Reference on LoanState */
+  loanState = LoanState;
+  /** is the current request is expired */
+  isRequestExpired = false;
+  /** Modal ref */
+  bsModalRef: BsModalRef;
+
+  // GETTER & SETTER =========================================================
+  /**
+   * Is the debug mode could be activated ?
+   * @returns True if the debug mode can be enabled and switched
+   */
+  get canUseDebugMode(): boolean {
+    return this._userService.user.isSystemLibrarian;
+  }
+
+  // CONSTRUCTOR & HOOKS ======================================================
+  /**
+   * Constructor
+   * @param _router - Router
+   * @param _serializer - UrlSerializer
+   * @param _userService - UserService
+   * @param _modalService - BsModalService
+   */
+  constructor(
+    private _router: Router,
+    private _serializer: UrlSerializer,
+    private _userService: UserService,
+    private _modalService: BsModalService
+  ){ }
+
+  /** OnInit hook */
+  ngOnInit() {
+    // Build related resource URL's
+    const itemUrlTree = this._router.createUrlTree(['records', 'items', 'detail', this.record.metadata.item.pid]);
+    this.itemUrl = this._serializer.serialize(itemUrlTree);
+    const patronUrlTree = this._router.createUrlTree(['records', 'patrons', 'detail', this.record.metadata.patron.pid]);
+    this.patronUrl = this._serializer.serialize(patronUrlTree);
+    // State bullet color
+    this._getBadgeStyle();
+    // Is request is expired
+    if ('request_expire_date' in this.record.metadata) {
+      const requestExpireDate = moment(this.record.metadata.request_expire_date);
+      this.isRequestExpired = moment().isSameOrAfter(requestExpireDate);
+    }
+  }
+
+  // COMPONENT FUNCTIONS ======================================================
+  /** Open transaction history logs dialog */
+  openTransactionHistoryDialog(loanPid: string): void {
+    const config = {
+      ignoreBackdropClick: false,
+      keyboard: true,
+      initialState: {
+        resourcePid: loanPid,
+        resourceType: 'loan'
+      }
+    };
+    this.bsModalRef = this._modalService.show(CirculationLogsComponent, config);
+    this.bsModalRef.content.dialogClose$.subscribe((value: boolean) => {
+      if (value) {
+        this.bsModalRef.hide();
+      }
+    });
+  }
+
+
+  // PRIVATE COMPONENT FUNCTIONS ==============================================
+
+  /** Define the bullet color. */
+  private _getBadgeStyle(): void {
+    switch (this.record.metadata.state) {
+      case LoanState.CREATED:
+      case LoanState.PENDING:
+      case LoanState.ITEM_AT_DESK:
+        this.stateClass = 'badge-info';
+        break;
+      case LoanState.ITEM_ON_LOAN:
+        this.stateClass = 'badge-success';
+        break;
+      case LoanState.ITEM_IN_TRANSIT_FOR_PICKUP:
+      case LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
+        this.stateClass = 'badge-warning';
+    }
+  }
```

Comment:
pipe ?

Reply:
Pleaaaassse. I will not writing a pipe for every small variable testing. Sure it will be great, but is it really necessary ? It will never used in any other component.